featurebyte/service/feature_readiness.py

```diff
@@ -121,38 +117,28 @@ async def update_feature_list_namespace(
         if feature_list_ids != document.feature_list_ids:
             update_dict["feature_list_ids"] = feature_list_ids
 
-        if document.default_version_mode == DefaultVersionMode.AUTO:
-            if feature_list_ids:
-                default_feature_list = await self._get_default_feature_list(feature_list_ids)
-                update_dict["default_feature_list_id"] = default_feature_list.id
-                update_dict["readiness_distribution"] = default_feature_list.readiness_distribution
-        else:
-            assert (
-                document.default_feature_list_id not in excluded_feature_list_ids
-            ), "default feature list should not be deleted"
-            default_feature_list = await self.feature_list_service.get_document(
-                document_id=document.default_feature_list_id
-            )
-            if default_feature_list.readiness_distribution != document.readiness_distribution:
-                # when feature readiness get updated and feature list namespace in manual default mode
-                update_dict["readiness_distribution"] = default_feature_list.readiness_distribution
+        if feature_list_ids:
+            default_feature_list = await self._get_default_feature_list(feature_list_ids)
+            update_dict["default_feature_list_id"] = default_feature_list.id
+            update_dict["readiness_distribution"] = default_feature_list.readiness_distribution
 
-        updated_document: Optional[FeatureListNamespaceModel] = document
         if update_dict:
-            updated_document = await self.feature_list_namespace_service.update_document(
+            await self.feature_list_namespace_service.update_document(
```

Comment:
curious: are these changes to split `update_document` into update->get required as part the version_mode changes?

Reply:
The main purpose of changing these is to make the pycharm type hinting happy (rather than returning an `Optional[FeatureModel]`, make it return `FeatureModel`). One of the major reason to use `Optional[FeatureModel]` is due to the GitDB (querying DB is very expensive). After dropping that use case, querying mongo here is relatively cheap.

Reply:
Before:
![Screenshot 2023-08-01 at 11 47 32 AM](https://github.com/featurebyte/featurebyte/assets/37915749/d54e14f2-bbaf-4c3d-a91c-2da3b9aae1b3)

After:
![Screenshot 2023-08-01 at 11 48 31 AM](https://github.com/featurebyte/featurebyte/assets/37915749/76499505-54fc-4d8d-9561-9f9e39e8d0be)

Reply:
ah gotcha, sounds good, thanks!